salt/utils/process.py

```diff
@@ -703,35 +698,32 @@ def kill_children(self, *args, **kwargs):
 
 
 class Process(multiprocessing.Process, NewStyleClassMixIn):
+    def __new__(cls, *args, **kwargs):
+        # We implement __new__ because we want to capture the passed in *args and **kwargs
```

Comment:
I'm still not a fan of this :) If feels like we are trying to be magic for no reason. I don't see the problem with having `__getstate__` and `__setstate__` explicitly defined where it's being used. It's more obvious what's going on that way. IMO.

Reply:
I know, and I'm still a fan of it.
It assures we, or anyone, never forgets, and removed duplicate boilerplate code to support pickling

Reply:
Also, we're already using a custom `multiprocessing.Process` class which alleviates on required boilerplate code, so, why is adding additional support to remove boilerplate code such an issue?